clippy_lints/src/methods/map_clone.rs

```diff
@@ -114,19 +114,17 @@ fn handle_path(
 ) {
     if let Some(path_def_id) = cx.qpath_res(qpath, arg.hir_id).opt_def_id()
         && cx.tcx.lang_items().get(LangItem::CloneFn) == Some(path_def_id)
-    {
         // The `copied` and `cloned` methods are only available on `&T` and `&mut T` in `Option`
         // and `Result`.
-        if let ty::Adt(_, args) = cx.typeck_results().expr_ty(recv).kind()
+        && let ty::Adt(_, args) = cx.typeck_results().expr_ty(recv).kind()
             && let args = args.as_slice()
             && let Some(ty) = args.iter().find_map(|generic_arg| generic_arg.as_type())
             && let ty::Ref(_, ty, Mutability::Not) = ty.kind()
             && let ty::FnDef(_, lst) = cx.typeck_results().expr_ty(arg).kind()
             && lst.iter().all(|l| l.as_type() == Some(*ty))
             && !should_call_clone_as_function(cx, *ty)
```

Comment:
Formatting of `if let` chains doesn't seem to work.

```suggestion
        && let ty::Adt(_, args) = cx.typeck_results().expr_ty(recv).kind()
        && let args = args.as_slice()
        && let Some(ty) = args.iter().find_map(|generic_arg| generic_arg.as_type())
        && let ty::Ref(_, ty, Mutability::Not) = ty.kind()
        && let ty::FnDef(_, lst) = cx.typeck_results().expr_ty(arg).kind()
        && lst.iter().all(|l| l.as_type() == Some(*ty))
        && !should_call_clone_as_function(cx, *ty)
```

Reply:
Indeed. Fixed.